docs/technical-overview.md

```diff
@@ -0,0 +1,250 @@
+
+
+## Principal components
+
+### Overview
+
+
+```plantuml
+@startuml
+node "Registry" {
+  [Registry API]
+  [Admin Webapp] --> [Registry API]
+}
+
+node "UAA" {
+  [User Auth API] --> [Registry API]
+}
+
+node "Gateway" {
+  [Gateway API] --> [Registry API]
+  [Podium Webapp] --> [Gateway API]
+}
+
+cloud {
+  [Github Config]
+  [Registry API] --> [Github Config]
+}
+
+rectangle Common [
+  podium-common
+  * Shared DTOs
+  * Request type and status enums
+  * Event definitions
+  * Exception classes
+  * etc
+]
+
+[Gateway API] ..> Common
+[User Auth API] ..> Common
+
+
+database "PostgreSQL" {
+  frame "Auth DB" as AuthDB {
+  }
+  frame "Request DB" as RequestDB {
+  }
+}
+
+rectangle "Auth ES Index" as AuthES
+AuthES -[hidden]-> AuthDB
+rectangle "Request ES Index" as RequestES
+RequestES -[hidden]-> RequestDB
+
+@enduml
+```
+
+
+### Registry
+
+The registry is a system administration component that plays a role of a service discovery hub and a configuration provider used by the other components of the Podium system.
+
+Registry frontend is an AngularJS webapp that displays the system health status, lists running component instances, shows configuration options & environment variables, and gives access to various metrics that might be useful to monitor the system status.
+The webapp fetches all data from the registry backend only, it does not interact with any other components.
+Since the webapp is not supposed to be accessible to end users and is not essential to run the Podium system; it is reasonable to assume that non-backward compatible changes are completely acceptable during the upgrade as long as a technical user can find metrics necessary for operations & maintenance.
+
+Supported NodeJS version: 8.12.0+.
+
+
+The registry Backend is a Spring Boot application, based on Netflix Eureka for service discovery and using Spring Cloud Config to fetch git-based configuration data. The API provided by the registry Backend is essential to run the Podium, but the concrete functionality can better be described in the context of the dependent components, as certain endpoints, e.g. `/config/*`, are not implemented explicitly as a part of the project — see Spring documentation for details.
+
+Current Podium Registry version: 1.0.4; supported JDK version: 8.
+
+
+### User Accounts & Authentication Server
+
+The UAA component is a Spring Boot application that defines user roles (Admin, Researcher etc), provides methods for user creation, password management, user search, authentication (username/password, OAuth2) etc.
+
+UAA server sends out emails to request email verification and send password reset links, to inform Podium administrators of user creation/registration etc.
+
+Special users that always exist: "admin", "system".
```

Comment:
`system`: yes. This is the name that is used when no username is available, e.g., for internal client based communication. The `admin` account is generated as one of the test accounts in development mode, but in principle any user can be promoted to be administrator (by an administrator).